sensing/autoware_pointcloud_preprocessor/src/downsample_filter/pickup_based_voxel_grid_downsample_filter_node.cpp

```diff
@@ -93,9 +93,14 @@ void PickupBasedVoxelGridDownsampleFilterComponent::filter(
   voxel_map.reserve(input->data.size() / input->point_step);
 
   constexpr float large_num_offset = 100000.0;
-  const float inverse_voxel_size_x = 1.0 / voxel_size_x_;
-  const float inverse_voxel_size_y = 1.0 / voxel_size_y_;
-  const float inverse_voxel_size_z = 1.0 / voxel_size_z_;
+  const float inverse_voxel_size_x = (voxel_size_x_ > 0) ? (1.0f / voxel_size_x_) : 0.0f;
+  const float inverse_voxel_size_y = (voxel_size_y_ > 0) ? (1.0f / voxel_size_y_) : 0.0f;
+  const float inverse_voxel_size_z = (voxel_size_z_ > 0) ? (1.0f / voxel_size_z_) : 0.0f;
+
```

Comment:
voxel_size_ is a float, so it should not be compared with an integer 0. Moreover, if it is 0.0 or less, the algorithm will not function properly. Therefore, the parameter should be set to ensure it never falls below 0.0, or an exception should be thrown to return or terminate the process.

Reply:
Thank you for your review. I fixed 
- `voxel_size_` to compares `0.0f`
- to set `max_inverse` if `voxel_size_` is `0.0f` or less.
  - This prevents integer overflow in the later static_cast operation. https://github.com/autowarefoundation/autoware.universe/blob/7826847ba3ad904811183257d7b076b169687622/sensing/autoware_pointcloud_preprocessor/src/downsample_filter/pickup_based_voxel_grid_downsample_filter_node.cpp#L121-L123
- to use error logging instead of throwing exceptions for practicality

Reply:
@yukkysaito 
I should have check it in the intialization phase :pray: 
I updated the code by adding validation for voxel size parameters that shuts down the node if invalid values (≤ 0) are detected